### ISSUE TITLE
The edgedns module rewrites the host's /etc/resolv.conf, resulting in slower access

### ISSUE THREAD

**Issue by Poorunga** [kubeedge/edgemesh#220]
**Working Principle of EdgeDNS**:

The edgedns is a submodule of edgemesh-agent, edgedns module will insert a nameserver into the /etc/resolv.conf file of the host, such as:

```
[root@ke-edge1 ~]# cat /etc/resolv.conf 
# Generated by NetworkManager
search openstacklocal
nameserver 169.254.96.16   <------  here
nameserver 100.125.3.250
nameserver 100.125.1.250
options single-request-reopen
```

After edged(a edgecore submodule) pulls up the container, it will write the first three nameservers of /etc/resolv.conf into /etc/resolv.conf in the pod. DNS requests in the pod, such as 'svc-name.namespace.svc.cluster.local', will be resolved by 'nameserver 169.254.96.16'.

PS: If edged.clusterDNS(in edgecore.yaml) is empty, then 'ClusterFirst' and 'ClusterFirstWithHostNet' have no effect and will run in 'Default' mode. Please refer to: https://kubernetes.io/zh/docs/concepts/services-networking/dns-pod-service/#pod-s-dns-policy

**Some Problems Happened**:
1. Some pods started before edgemesh-agent, their /etc/resolv.conf cannot be written into the 'nameserver 169.254.96.16', causing domain name resolution to fail
2. The domain name resolution of the host itself will be affected and become particularly slow

**How to Solve These Problems**
- [x] The edgedns no longer write nameserver to the host's /etc/resolv.conf
- [x] Update edgemesh document(the user needs to write '169.254.96.16' to the edged.clusterDNS (in edgecore.yaml))
- [x] The edgedns module needs to be able to support the resolution of external domain names, such as: www.kubeedge.io ...

**Comment by srnbckr** [comment]
This seems to break the DNS functionality of pods spawned by KubeEdge Sedna: For "normal" pods started with i.e. kubectl, the edged service writes the correct nameserver in the /etc/resolv.conf of the pod, but Sedna pods started by i.e. a FederatedLearningJob crd don't get the update in /etc/resolv.conf and therefore can't reach cloud services.

**Comment by Poorunga** [comment]
@srnbckr  FederatedLearningJob use hostNetwork, so you need config your job yaml like "dnsPolicy: ClusterFirstWithHostNet"

You can refer to: https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-dns-policy

**Comment by Poorunga** [comment]
bug fixed in PR #245
